### ISSUE TITLE
Timedelta doesn't support `days=None`

### ISSUE THREAD

**Issue by Buried-In-Code** [Buried-In-Code/Simyan#36]
Glancing real quick at this, the `insert` method for the `SqliteCache` class was:
```
    def insert(self, query: str, response: str):
        expiry = datetime.now() + timedelta(days=self.expiry)
        self.cur.execute(
            "INSERT INTO queries(query, response, expiry) VALUES(?, ?, ?);",
            (query, json.dumps(response), expiry.strftime("%Y-%m-%d")),
        )
        self.con.commit()
```
but with this pull request the `expiry` parameter can be `None`, so it probably needs to be changed to something like this to head off any potential bugs. I'm not on my dev machine, so you should **definitely** double check this solution. ;)

```
    def insert(self, query: str, response: str):
        if self.expiry:
            expiry = datetime.now() + timedelta(days=self.expiry)

            self.cur.execute(
                "INSERT INTO queries(query, response, expiry) VALUES(?, ?, ?);",
                (query, json.dumps(response), expiry.strftime("%Y-%m-%d")),
            )
        else:
            self.cur.execute(
                "INSERT INTO queries(query, response) VALUES(?, ?);",
                (query, json.dumps(response)),
            )
        self.con.commit()
```

_Originally posted by @bpepple in https://github.com/Buried-In-Code/Simyan/pull/34#issuecomment-917794945_

**Comment by Buried-In-Code** [comment]
I checked on my setup and it does through an error, I guess it was missed as I still had the cache so it didn't try doing an insert

**Comment by Buried-In-Code** [comment]
Closed in #37
